core/src/main/java/org/radargun/stages/cache/background/AbstractLogLogic.java

```diff
@@ -219,28 +230,50 @@ protected void afterRollback() {
       delayedRemoves.clear();
    }
 
-   protected void afterCommit() {
+   protected boolean afterCommit() {
       boolean inTransaction = false;
       try {
+         Map<Long, Integer> delayedRemoveAttemptsMap = new HashMap<>();
          while (!stressor.isTerminated()) {
             try {
-               if (inTransaction) {
+               if (inTransaction || ongoingTx != null) {
```

Comment:
Looks like we don't need the inTransaction variable at all if ongoingTx != null signalizes that we really are in tx. Or is there any reasonable case when inTransaction == false && ongoingTx != null?